modules/installation-gcp-permissions.adoc

```diff
@@ -23,32 +24,63 @@ install {product-title}. To deploy an {product-title} cluster, the service
 account requires the following permissions. If you deploy your cluster into an existing VPC, the service account does not require certain networking permissions, which are noted in the following lists:
 
 .Required roles for the installation program
-* Compute Admin
-* Security Admin
-* Service Account Admin
-* Service Account User
-* Storage Admin
+[cols="3,7"options="header"]
+|===
+|Role|Description
+
+|Compute Admin
+|Gives full control of all compute resources. This role is needed by the service account to manage virtual machine instances. {product-title} is a self-managed platform that requires permission to create cloud infrastructure to automatically scale and failover during normal operation.
+
+|Security Admin
+|Allow the service account to create, modify, and delete firewall rules and SSL certificates within the GCP project.
+
+|Service Account Admin
+|Creates and manages service accounts.
+
+|Service Account User
+|Runs operations as the service account.
+
+|Storage Admin
+|Grants full control of storage objects and buckets, and is needed to help provide persistent storage to the cluster.
+|===
 
 .Required roles for creating network resources during installation
-* DNS Administrator
+[cols="3,7"options="header"]
+|===
+|Role|Description
+
+|DNS Administrator
+|Provides the service account read-write access to all Cloud DNS resources. To install {product-title}, your GCP account must have a dedicated public hosted zone in the same project that you host the cluster. This zone must be authoritative for the domain. The DNS service provides cluster DNS resolution and name lookup for external connections to the cluster.
+|===
 
 ifdef::template[]
 .Required roles for user-provisioned GCP infrastructure
-* Deployment Manager Editor
-* Service Account Key Admin
+[cols="3,7"options="header"]
+|===
+|Role|Description
+
+
+|Deployment Manager Editor
+|?
+
+|Service Account Key Admin
+|?
```

Comment:
@jstuever can you provide some information on why the `Deployment Manager Editor` and `Service Account Key Admin` GCP roles are required for a UPI installation?

Reply:
The UPI workflow uses the deployment manager editor to create the deployments, which is the basis of the UPI workflow. This would be each time `gcloud deployment-manager deployments create` is called. The service account key admin is necessary to add a key to the service account with `gcloud iam service-accounts keys create service-account-key.json`. This key is then used to create a secure signed-url for the boostrap ignition with `gsutil signurl`.